### PR TITLE
Adding unit tests for each tuning type

### LINKED ISSUE #74: Add unit tests for tuning/sft_trainer.py
## Is your feature request related to a problem? Please describe.

Add unit tests for the tuning/sft_trainer.py module.

## Describe the solution you'd like

If possible, each function should have test cases in the file tests/test_sft_trainer.py
and pass test with the command `pytest tests/test_sft_trainer.py`

## Additional context

Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by tharapalanivel**
### Description of the change

Adding unit tests for `pt` and `lora` tuning method using dummy model
### Related issue number

Closes #74 

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [ ] I have ensured all unit tests pass

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 numpy
 accelerate>=0.20.3
 packaging
-transformers>=4.34.1
+transformers>=4.34.1,<4.38.0
```
[the remark this answers is not in the document]
Comment (fabianlim) [reply requirements.txt:4]: the reason you are seeing this error is because `TrainingArguments` is not initialized properly. In the code, we inherit from the dataclass, but we did not call `.__post_init__.` This PR #53 fixes this issue.

**Comment by fabianlim** [comment]
Not sure if its a good idea to commit binary files that are 17MB large into a git repo.

File: tests/test_sft_trainer.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    "per_device_eval_batch_size": 4,
+    "gradient_accumulation_steps": 4,
+    "learning_rate": 0.00001,
+    "weight_decay": 0,
+    "warmup_ratio": 0.03,
+    "lr_scheduler_type": "cosine",
+    "logging_steps": 1,
+    "include_tokens_per_second": True,
+    "packing": False,
+    "response_template": "\\n###Response:",
```
Comment (anhuong) [inline tests/test_sft_trainer.py:44]: This should be `### Label` instead of Response and note the space. This matches the twitter dataset input

File: tests/test_sft_trainer.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        model_args, data_args, training_args, tune_config = causal_lm_train_kwargs(
+            HAPPY_PATH_KWARGS
+        )
+        sft_trainer.train(model_args, data_args, training_args, tune_config)
+        _validate_training(tempdir)
+
+
+def _validate_training(tempdir):
+    assert any(x.startswith("checkpoint-") for x in os.listdir(tempdir))
+    loss_file_path = "{}/train_loss.jsonl".format(tempdir)
```
Comment (anhuong) [inline tests/test_sft_trainer.py:84]: perhaps update to `train_loss_file_path` to distinguish between eval loss and train loss

File: tests/helpers.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# limitations under the License.
+
+# Third Party
+import transformers
+
+# Local
+from tuning.config import configs, peft_config
+
+
+def causal_lm_train_kwargs(train_kwargs):
```
Comment (anhuong) [inline tests/helpers.py:22]: This has come up a few times now, that this code is duplicated in launch_training script and now is duplicated in tests. Feels like it may be worthwhile to refactor the argument parsing into its own method that can be tested separated in unit tests and called in both places.

File: tests/helpers.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        prompt_tuning_config,
+    ) = parser.parse_dict(train_kwargs, allow_extra_keys=True)
+
+    # TODO: target_modules doesn't get set probably due to the way dataclass handles
+    # mutable defaults, needs investigation on better way to handle this
+    setattr(
+        lora_config,
+        "target_modules",
+        lora_config.__dataclass_fields__.get("target_modules").default_factory(),
+    )
```
Comment (anhuong) [inline tests/helpers.py:47]: Can you elaborate on this error?
